src/docfx/validation/CustomRuleProvider.cs

```diff
@@ -48,6 +52,17 @@ public bool IsEnable(FilePath filePath, CustomRule customRule, string? moniker =
                 return false;
             }
 
+            if (customRule.Tags != null && customRule.Tags.Contains("SEO"))
+            {
+                var metadata = _metadataProvider.GetMetadata(ErrorBuilder.Null, filePath).RawJObject;
+                var noindex = metadata.TryGetValue<JValue>("robot", out var robotValue) && robotValue.Value is string robot &&
```

Comment:
prefer `UserMetadata.Robots`

Reply:
`UserMetadata.Robots` is null here. And I find some code like this:
```
_metadataValidator.ValidateMetadata(errors, userMetadata.RawJObject, file);
```

Reply:
The test cases is wrong :)

Reply:
good catch!😀